lib/services/android-project-service.ts

```diff
@@ -253,7 +255,63 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
+				hasAndroidManifestInAppResources = this.$fs.exists(originalAndroidManifestFilePath).wait(),
+				shouldExtractDefaultManifest = !hasAndroidManifestInAppResources;
+
+			if(hasAndroidManifestInAppResources) {
+				// Check if the AndroidManifest in app/App_Resouces is the correct one
+				// Use a real magic to detect if this is the correct file, by checking some mandatory strings.
+				let fileContent = this.$fs.readText(originalAndroidManifestFilePath).wait(),
+					isFileCorrect = !!(~fileContent.indexOf("android:minSdkVersion") && ~fileContent.indexOf("android:targetSdkVersion")
+									&& ~fileContent.indexOf("uses-permission") && ~fileContent.indexOf("<application")
+									&& ~fileContent.indexOf("<activity") && ~fileContent.indexOf("<intent-filter>")
+									&& ~fileContent.indexOf("android.intent.action.MAIN") && ~fileContent.indexOf("com.tns.ErrorReportActivity")
+									&& ~fileContent.indexOf("android:versionCode"));
+
+				this.$logger.trace(`Existing ${this.platformData.configurationFileName} is ${isFileCorrect ? "" : "NOT "}correct.`);
+
+				if(!isFileCorrect) {
+					shouldExtractDefaultManifest = true;
+					let newNameForOriginalManifest = "old." + this.platformData.configurationFileName,
+						newPathForOriginalManifest = path.join(path.dirname(originalAndroidManifestFilePath), newNameForOriginalManifest);
+					this.$logger.warn(`Your ${this.platformData.configurationFileName} in app/App_Resources/Android will be replaced by the default one from hello-world template.`);
+					this.$logger.printMarkdown(`The original file will be moved to \`${newPathForOriginalManifest}\`. Merge it **manually** with the new \`${this.platformData.configurationFileName}\` in your app/App_Resources/Android.`);
+					shell.mv(originalAndroidManifestFilePath, newPathForOriginalManifest);
+				}
+			}
+
+			if(shouldExtractDefaultManifest) {
+				let defaultTemplatePath = this.$projectTemplatesService.defaultTemplatePath.wait();
+				let templateAndroidManifest = path.join(defaultTemplatePath, constants.APP_RESOURCES_FOLDER_NAME, this.$devicePlatformsConstants.Android, this.platformData.configurationFileName);
+				if (this.$fs.exists(templateAndroidManifest).wait()) {
+					this.$logger.trace(`${originalAndroidManifestFilePath} is missing. Upgrading the source of the project with one from the new project template. Copy ${templateAndroidManifest} to ${originalAndroidManifestFilePath}`);
+					try {
+						this.$fs.copyFile(templateAndroidManifest, originalAndroidManifestFilePath).wait();
+					} catch(e) {
+						this.$logger.trace(`Copying template's ${this.platformData.configurationFileName} failed. `, e);
+						return;
+					}
+				} else {
+					this.$logger.trace(`${originalAndroidManifestFilePath} is missing but the template ${templateAndroidManifest} is missing too, can not upgrade ${this.platformData.configurationFileName}.`);
```

Comment:
What can the user do to fix this scenario? Can we point them in the right direction with a message?